core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1142,11 +1143,22 @@ private boolean containsAggregations(SqlNode node) {
         if (call.getOperator() instanceof SqlAggFunction) {
           return true;
         } else {
-          for (SqlNode operand : call.getOperands()) {
-            if (containsAggregations(operand)) {
-              return true;
-            }
-          }
+          return containsAggregations(Arrays.asList(call.getOperands()));
+        }
+      } else if (node instanceof SqlCase) {
```

Comment:
I'm not sure this covers every other missing case here. What about `with` which also extends `SqlCall`?
How about casting to a `SqlCall` and checking the `getOperandList` & `getOperator` methods?